src/performance/performance.ts

```diff
@@ -32,14 +31,12 @@ export class AngularFirePerformance {
 
     this.performance = of(undefined).pipe(
       switchMap(() => isPlatformBrowser(platformId) ? zone.runOutsideAngular(() => import('firebase/performance')) : EMPTY),
-      switchMap(() => import('@firebase/performance')),
-      tap(perf => perf.registerPerformance && perf.registerPerformance(firebase as any)),
```

Comment:
Glad to see these aren't needed. I did some testing to see if they were causing issues, but they weren't... though I'm not sure if they were doing anything useful either :)

Reply:
Yeah these register calls aren't great... it's weird and they're private APIs, so the JS SDK team can change them at any point.

I've found that I've needed them for some SSR cases but it seems marking everything as an external in angular.json makes it so they aren't always needed. In either case, perf mon is only browser, so I can drop this. I believe the root problem is due to ngcc failing when it hits the CJS version of Firebase. It's somehow messing with the side-effects of the import. We've tried to address a couple times but keep running into issues. That said, we have a tree-shakable version of the JS SDK on the way https://github.com/firebase/firebase-js-sdk/blob/master/docs-exp/index.md that should play much much better with Angular; so I don't think it's worth spending much time trying to solving the CJS / tree-shake problems with NG10 & the namespaced JS SDK.

Reply:
Great news on the tree-shakable JS SDK 👍  Sounds like there will be plenty of refactoring later this year or next year 😁